velox/expression/Expr.cpp

```diff
@@ -259,6 +289,110 @@ bool isFlat(const BaseVector& vector) {
 
 } // namespace
 
+bool MutableRemainingRows::deselectNonErrorNulls(
```

Comment:
MutableRemainingRows is a class that should only concern itself with modifying the remaining rows, this function however mutates both the context and 'errors' vector which can get quite confusing as this side effect can be hidden while reading code. I have a recommendation for changes to Expr::evalArgsDefaultNulls() that would simplify the logic there and help us get rid of the side effect i mentioned above. Please take a look.